File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Events/CalcDamageValueMessage.ts
import { IMessage } from "../../../../Core/ObserverMessageKit";
import DamageInfo from "../DataDefine/DamageInfo";
import AbstractCreature from "../Unit/AbstractCreature";

//计算伤害
export class CalcDamageValueMessage implements  IMessage{
    public static readonly Id = "CalcDamageValue";
    public constructor(public info:DamageInfo,public source:AbstractCreature,public target: AbstractCreature) {
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Event/Ids/CalcDamageValueMessage.cs
using ModTheSpire.DataDefine;
using ModTheSpire.Unit;

namespace ModTheSpire.Event.Ids
{
    ///计算伤害
    public class CalcDamageValueMessage : IEventHandle
    {
        public DamageInfo DamageInfo;
        public AbstractCreature Source;
        public AbstractCreature Target;

        public CalcDamageValueMessage(DamageInfo damageInfo, AbstractCreature source, AbstractCreature target)
        {
            DamageInfo = damageInfo;
            Source = source;
            Target = target;
        }
    }
}